src/documents/tests/test_api.py

```diff
@@ -2369,6 +2369,55 @@ def test_create_note(self):
 
         self.assertEqual(resp_data["note"], "this is a posted note")
 
+    def test_notes_permissions_aware(self):
+        """
+        GIVEN:
+            - Existing document owned by user2 but with granted view perms for user1
+        WHEN:
+            - API request is made by user1 to add a note or delete
+        THEN:
+            - Notes are neither created nor deleted
+        """
+        user1 = User.objects.create_user(username="test1")
+        user1.user_permissions.add(*Permission.objects.filter(codename="add_document"))
+        user1.user_permissions.add(*Permission.objects.filter(codename="add_note"))
+        user1.user_permissions.add(*Permission.objects.filter(codename="delete_note"))
+        user1.save()
+
+        user2 = User.objects.create_user(username="test2")
+        user2.save()
+
+        doc = Document.objects.create(
+            title="test",
+            mime_type="application/pdf",
+            content="this is a document which will have notes added",
+        )
+        doc.owner = user2
+        doc.save()
+
+        assign_perm("view_document", user1, doc)
+
+        self.client.force_authenticate(user1)
+
+        resp = self.client.post(
+            f"/api/documents/{doc.pk}/notes/",
+            data={"note": "this is a posted note"},
+        )
+        self.assertEqual(resp.status_code, status.HTTP_403_FORBIDDEN)
+
+        note = Note.objects.create(
+            note="This is a note.",
+            document=doc,
+            user=user2,
+        )
+
+        response = self.client.delete(
+            f"/api/documents/{doc.pk}/notes/?id={note.pk}",
+            format="json",
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
     def test_delete_note(self):
```

Comment:
Do you want to add testing for the delete that is permissions aware?

Reply:
TBH, I thought that this did (user1 tries to delete a note from user2's doc)
https://github.com/paperless-ngx/paperless-ngx/blob/49b915eb3300e11a087e3a7cf6283a29ec1c7996/src/documents/tests/test_api.py#L2408-L2419

My pytest setup seems randomly broken at the moment so Im a little hobbled

Reply:
TBH, I thought that this did (user1 tries to delete a note from user2's doc)
https://github.com/paperless-ngx/paperless-ngx/blob/49b915eb3300e11a087e3a7cf6283a29ec1c7996/src/documents/tests/test_api.py#L2408-L2419

My pytest setup seems randomly broken at the moment so Im a little hobbled

Reply:
It looks like `view_document` doesn't allow a user to view the notes on a document.  I'm not sure if it should.  But I think the delete gets blocked earlier up the chain and doesn't reach this.

Reply:
Cheers stumpy, got me on the right track! Should be sorted (as for pytest, I'll live to fight that one another day 🙃)